### PR TITLE
[FeatureRequest: Issue #311] Support Terraform versions greater than or equal to 1.0.0

### LINKED ISSUE #311: Support Terraform versions >= 1.0.0
## Is your feature request related to a problem?
The current install script does not support using Terraform (TF) versions greater than or equal to 1.0.0.

## Describe the solution you'd like
I'd like to be able to use modern versions of TF when contributing to the project for tasks like running tests and building the provider.

## Acceptance criteria

- All unit and integration tests should proceed when TF >= 1.0.0 is installed
- Building the provider should proceed when TF >= 1.0.0 is installed

## Describe alternatives you've considered
None

## Additional context
I'll submit a PR and reference this issue.

## Checklist (for admin only)

**Don't forget to go through the checklist to make sure the issue is created properly**:

- [ ] I have added a corresponding label (feature request) to the issue (right side menu)
- [ ] I have added this issue to the 'API Terraform Provider' GitHub project (right side menu)

### PR DESCRIPTION
**Pull request by bradthebuilder**
## What problem does this Pull Request solve?

Please link to the issue number here (issue will be closed when PR is merged): Closes #311

## Type of change

What type of change does your code introduce to the provider? Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] New feature (change that adds new functionality)
- [ ] Bug-fix (change that fixes current functionality)
- [ ] Tech debt (enhances the current functionality)
- [ ] New release (pumps the version)

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have made sure code compiles correctly and all tests are passing by running `make test-all`
- [ ] I have added/updated necessary documentation (if appropriate)
- [x] I have added the following info to the title of the PR (pick the appropriate option for the type of change). This is important because the release notes will include this information.
  - [x] Feature Request: PRs related to feature requests should have in the title `[FeatureRequest: Issue #X] <PR Title>`
  - [ ] Bug Fixes: PRs related to bug fixes should have in the title `[BugFix: Issue #X] <PR Title>`
  - [ ] Tech Debt: PRs related to technical debt should have in the title `[TechDebt: Issue #X] <PR Title>` 
  - [ ] New Release: PRs related to a new release should have in the title `[NewRelease] vX.Y.Z`

## Commentary
All unit and integration tests pass when using TF versions 1.0.1 and 1.0.7 on my system. I'm also able to successfully run make build. I've tested the built provider against my Antsle and it successfully created Antlets using my TF configs.

### Unit Tests Passing
unit_tests_passing

### Integration Tests Passing
integration_tests_passing

Targeting your master branch per your request.

There is also a single commit addressing what I thought was a typo in some documentation.

## Checklist for Admins
- [ ] Label is populated
- [ ] PR is assigned to the corresponding project
- [ ] PR has at least 1 reviewer and 1 assignee

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -104,16 +104,20 @@ function setupInstallationPath(){
...
     eerror "OpenAPI Terraform provider no longer supports versions of Terraform less than v0.12.0"
     exit 1
-  elif [ $TF_MINOR_VERSION -eq 12 ]
+  elif [ $TF_MINOR_VERSION -eq 12 ] && [ $TF_MAJOR_VERSION -eq 0 ]
   then
     einfo "Installing provider based on Terraform v0.12.* plugin installation instructions: https://www.terraform.io/docs/plugins/basics.html#installing-plugins"
     TF_PROVIDER_PLUGIN_NAME="${TF_PROVIDER_BASE_NAME}${PROVIDER_NAME}_v${PLUGIN_VERSION}"
-  elif [ $TF_MINOR_VERSION -ge 13 ]
+  else
   then
```
Comment (dikhan) [inline scripts/install.sh:121]: I believe this `then` needs to be removed, getting an error now when running the script:

````
./scripts/install.sh: line 121: syntax error near unexpected token `then'
./scripts/install.sh: line 121: `  then'
````

**Review by dikhan — CHANGES_REQUESTED** [review]
Minor code update but otherwise looks good. Once this is done, as discussed I will go ahead and create another PR cherry picking your commits (so the pipeline is able to run all the tests) and add you as reviewer.

File: scripts/install.sh (hunk 1 above)
Comment (bradthebuilder) [reply scripts/install.sh]: Sorry about that. Removed that line; `make build -p <PROVIDER_NAME>` works now.

**Comment by dikhan** [comment]
As previously discussed, closing this in favour of PR-314

**Pull request closed without merging**